src/ethereum/prague/vm/precompiled_contracts/bls12_381/__init__.py

```diff
@@ -301,26 +302,13 @@
 MULTIPLIER = Uint(1000)
 
 
-def bytes_to_g1(
-    data: Bytes,
+@lru_cache(maxsize=128)
```

Comment:
Would this pose a DoS vector if it were implemented this way in a production client? Like could an attacker choose values so that the fast path is never/rarely executed?

If so, it isn't a deal-breaker, but we should add a comment to that effect here.

Reply:
Yes and I think that's exactly what `benchmark` tests are meant to test against, trending cache hits to zero while remaining inside constraints. I think a comment is well warranted 👍🏼